tools/launch.py

```diff
@@ -0,0 +1,135 @@
+        ip, _ = hosts[int(i / client_count_per_machine)]
+        cmd = client_cmd.replace('node_rank=0', 'node_rank='+str(i))
+        cmd = 'cd ' + str(args.workspace) + '; ' + cmd
+        execute_remote(cmd, ip)
+
+    # use ctl+C to exit
+    while True:
+        time.sleep(10)
+
+def server_command(args):
+    server_cmd = 'python3 ' + str(args.exe_file) + ' --server'
+    server_cmd = server_cmd + ' --graph-name ' + str(args.graph_name)
+    server_cmd = server_cmd + ' --num-client ' + str(args.num_client)
+    server_cmd = server_cmd + ' --conf_path ' + str(args.conf_path)
+    server_cmd = server_cmd + ' --ip_config ' + str(args.ip_config)
+    return server_cmd
+
+def client_command(args, udf_args):
+    client_cmd = 'python3 -m torch.distributed.launch'
+    client_cmd = client_cmd + ' --nproc_per_node=' + str(args.nproc_per_node)
+    client_cmd = client_cmd + ' --nnodes=' + str(args.nnodes)
+    client_cmd = client_cmd + ' --node_rank=' + str(0)
+    client_cmd = client_cmd + ' --master_addr=' + str(args.master_addr)
+    client_cmd = client_cmd + ' --master_port=' + str(args.master_port)
+    client_cmd = client_cmd + ' ' + str(args.exe_file)
+    client_cmd = client_cmd + ' --graph-name ' + str(args.graph_name)
+    client_cmd = client_cmd + ' --ip_config ' + str(args.ip_config)
+    client_cmd = client_cmd + ' ' + udf_args
+    return client_cmd 
+
+def main():
+    parser = argparse.ArgumentParser(description='Launch a distributed job')
+    ########## common args ####################
+    parser.add_argument('--kill',  action='store_true',
+                        help='kill all the python processes on cluster')
+    parser.add_argument('--workspace', type=str,
+                        help='Path of user workspace')
+    parser.add_argument('--ip_config', type=str, default='ip_config.txt',
+                        help='The file for IP configuration')
+    parser.add_argument('--exe_file', type=str,
+                        help="Python executable file")
+    parser.add_argument('--server', action='store_true',
+                        help='wether this is a server')
+    parser.add_argument('--graph-name', type=str, 
+                        help='graph name')
+    parser.add_argument('--num-client', type=int, 
+                        help='Total number of clients on cluster')
+    parser.add_argument('--conf_path', type=str, 
+                        help='The path to the partition config file')
+    ####### Pytorch distributed ################
+    parser.add_argument('--nproc_per_node', type=int, default=1,
+                        help='Number of processes per node')
+    parser.add_argument('--nnodes', type=int, 
+                        help='Total number of nodes')
+    parser.add_argument('--master_addr', type=str, 
+                        help='IP address of master node')
+    parser.add_argument('--master_port', type=int, default=1250,
+                        help='Port of master node')
```

Comment:
we don't need to provide `master_add` or `master_port`. we can determine them from ip_config